src/app/services/schema/app-schema.service.ts

```diff
@@ -637,65 +639,74 @@ export class AppSchemaService {
       }));
   }
 
-  private handleSchemaSubQuestions(payload: CommonSchemaAddControl, newFormControl: CustomUntypedFormControl): void {
-    const {
-      schema, isNew, path, subscription, formGroup, config, isParentImmutable,
-    } = payload;
+  private handleSchemaSubQuestions({
+    schema,
+    isNew,
+    path,
+    subscription,
+    formGroup,
+    config,
+    isParentImmutable,
+  }: CommonSchemaAddControl, newFormControl: CustomUntypedFormControl): void {
+    if (!schema.subquestions) {
+      return;
+    }
 
-    if (schema.subquestions) {
-      schema.subquestions.forEach((subquestion) => {
-        subscription.add(
-          this.addFormControls({
-            isNew,
-            path,
-            chartSchemaNode: subquestion,
-            formGroup,
-            config,
-            isParentImmutable: !!schema.immutable || isParentImmutable,
-          }),
-        );
+    for (const subquestion of schema.subquestions) {
+      subscription.add(
+        this.addFormControls({
+          isNew,
+          path,
+          chartSchemaNode: subquestion,
+          formGroup,
+          config,
+          isParentImmutable: !!schema.immutable || isParentImmutable,
+        }),
+      );
 
-        const formField = (formGroup.controls[subquestion.variable] as CustomUntypedFormField);
-        if (!formField.hidden$) {
-          formField.hidden$ = new BehaviorSubject<boolean>(false);
-        }
-        if (newFormControl.value === schema.show_subquestions_if) {
-          formField.hidden$.next(false);
-          formField.enable();
-        } else {
-          formField.hidden$.next(true);
-          formField.disable();
+      const formField = formGroup.controls[subquestion.variable] as CustomUntypedFormField;
+      this.initializeFormField(formField, newFormControl.value, schema, subquestion, isNew, isParentImmutable);
+    }
+
+    subscription.add(newFormControl.valueChanges.subscribe((value) => {
```

Comment:
In continuation of my earlier comment, maybe adding all the subscriptions can be a separate method call. This might not be possible depending on the context so take it with a grain of salt.